dotCMS/src/integration-test/java/com/dotcms/vanityurl/business/VanityUrlAPITest.java

```diff
@@ -60,271 +61,255 @@ public static void prepare() throws Exception {
-
-            Assert.assertNotNull(vanity1);
-            Assert.assertNotEquals(VanityUrlAPI.CACHE_404_VANITY_URL,vanity1.getInode());
-            Assert.assertEquals(title, vanity1.getTitle());
-            Assert.assertEquals(site, vanity1.getSite());
-            Assert.assertEquals(uri, vanity1.getURI());
-            Assert.assertEquals(forwardTo, vanity1.getForwardTo());
-            Assert.assertEquals(action, vanity1.getAction());
-
-        } catch (DotDataException | DotSecurityException e) {
-            e.printStackTrace();
-            fail();
-        } finally {
-            try {
-                if (contentlet1 != null) {
-                    contentletAPI.destroy(contentlet1, user, false );
-                }
-            } catch (Exception e) {
-                Logger.error(this.getClass(), "Error cleaning up Vanity Url Links");
-            }
-        }
+    public void test_that_regex_does_not_catch_regular_strings() throws Exception {
+
+
+        final String title = "VanityURL" + System.currentTimeMillis();
+        final String site = defaultHost.getIdentifier();
+        final String uri = "/saftey" + System.currentTimeMillis();
+        final String forwardTo = "/products/";
+        final int action = 301;
+        final int order = 1;
+        
+        final String requestUir=uri + "/health/preventing-injuries-illnesses/get-started-with-safety-health/chemical-safety-basics";
+
+       
+        // save and publish the vanity
+        filtersUtil.publishVanityUrl(filtersUtil.createVanityUrl(title, site, uri,
+                        forwardTo, action, order, defaultLanguage.getId()));
+        
+
+        
+        Optional<CachedVanityUrl> vanityResolved = vanityUrlAPI.resolveVanityUrl(requestUir, defaultHost, defaultLanguage);
+
+        // does not match /saftey
+        assert(!vanityResolved.isPresent());
+        
+        
+        // add a regex, which should now match
+        String newUrl = uri + ".*";
+        
+        // save and publish the vanity
+        filtersUtil.publishVanityUrl(filtersUtil.createVanityUrl(title, site, newUrl,
+                        forwardTo, action, order, defaultLanguage.getId()));
+        
+        
+        
+        vanityResolved = vanityUrlAPI.resolveVanityUrl(requestUir, defaultHost, defaultLanguage);
+
+        // DOES match /saftey.*
+        assert(vanityResolved.isPresent());
+        
+
     }
 
+    
+    
+    
+    
+    
+    
+    
+    
+    
     /**
      * This test creates a vanity url, gets it so it's cached.
      * Then modify the uri, and checks if the old one is removed from cache.
      * @throws DotSecurityException
      * @throws DotDataException
      */
     @Test
-    public void updateUriVanityURLcheckCacheTest() throws DotDataException, DotSecurityException {
-        long currentTime = System.currentTimeMillis();
+    public void vanities_resolve_correctly_after_url_has_been_updated() throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)